src/test/java/com/djaytan/bukkit/slf4j/BukkitLoggerFactoryTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * The MIT License
+ * Copyright © 2023 Loïc DUBOIS-TERMOZ (alias Djaytan)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.djaytan.bukkit.slf4j;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Field;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mock.Strictness;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.Logger;
+
+@ExtendWith(MockitoExtension.class)
+final class BukkitLoggerFactoryTest {
+
+  private static final String LOGGER_NAME = "test-logger-name";
+
+  @Mock(strictness = Strictness.LENIENT)
+  private java.util.logging.Logger sampleLogger;
+
+  private final BukkitLoggerFactory bukkitLoggerFactory = new BukkitLoggerFactory();
+
+  @BeforeEach
+  void beforeEach() {
+    when(sampleLogger.isLoggable(Level.INFO)).thenReturn(true);
+  }
+
+  @AfterEach
+  void tearDown() throws Exception {
+    Field staticLoggerField = BukkitLoggerFactory.class.getDeclaredField("staticLogger");
+    staticLoggerField.setAccessible(true);
+    staticLoggerField.set(null, null);
+  }
```

Comment:
The static field must be cleaned up at the end of each test because the class remains loaded between tests executions.

The usage of a static field in the production class has been directly questionned, but no trivial alternative has been found... And well in fact we are able to test the class without much headaches since the field is not `field`. Otherwise the story would be different.
Furthermore keeping the production code as is avoid making a breaking change while keeping a very simple way to setup the Bukkit logger on the client code side.